lib/src/modules/macho/mod.rs

```diff
@@ -264,6 +266,71 @@ fn has_rpath(ctx: &ScanContext, rpath: RuntimeString) -> Option<bool> {
     Some(false)
 }
 
+/// Returns an md5 hash of the dylibs designated in the mach-o binary
+#[module_export]
+fn dylib_hash(ctx: &mut ScanContext) -> Option<RuntimeString> {
+    let macho = ctx.module_output::<Macho>()?;
+    let mut md5_hash = Md5::new();
+    let mut dylibs_to_hash = &macho.dylibs;
+
+    // if there are not any dylibs in the main Macho, the dylibs of the nested file should be hashed
+    if dylibs_to_hash.is_empty() && !macho.file.is_empty() {
+        dylibs_to_hash = &macho.file[0].dylibs;
+    }
+
+    // we need to check again as the nested file dylibs could be empty too
+    if dylibs_to_hash.is_empty() {
+        return None;
+    }
+
+    let dylibs_to_hash: String = dylibs_to_hash
+        .iter()
+        .map(|d| {
+            std::string::String::from_utf8(d.name.clone().unwrap())
```

Comment:
If `d.name` is guaranteed to be `Some`, why make it optional?

Reply:
The generated Macho for the `macho.proto` keeps those fields as options even when the `macho.proto` marks them as `required` I believe.

Reply:
![image](https://github.com/VirusTotal/yara-x/assets/6127011/f088dfb1-fdc5-4f29-804a-898c8f863f1e)
![image](https://github.com/VirusTotal/yara-x/assets/6127011/fe34277b-f2e2-4117-b402-232e950ae2a4)